backend/api/Areas/Project/Controllers/ReportController.cs

```diff
@@ -0,0 +1,171 @@
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    [ApiVersion("1.0")]
+    [Area("projects")]
+    [Route("v{version:apiVersion}/[area]/reports")]
+    [Route("[area]/reports")]
+    public class ReportController : ControllerBase
+    {
+        #region Variables
+        private readonly IPimsService _pimsService;
+        private readonly IMapper _mapper;
+        private readonly PimsOptions _options;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a ReportController class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="pimsService"></param>
+        /// <param name="mapper"></param>
+        public ReportController(IOptionsMonitor<PimsOptions> options, IPimsService pimsService, IMapper mapper)
+        {
+            _options = options.CurrentValue;
+            _pimsService = pimsService;
+            _mapper = mapper;
+        }
+        #endregion
+
+        #region Endpoints
+        /// <summary>
+        /// Get the project report for the specified 'id'.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id:int}")]
+        [HasPermission(Permissions.AdminProjects)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ProjectReportModel), 200)]
+        [ProducesResponseType(typeof(ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] { "project" })]
+        public IActionResult GetProjectReport(int id)
+        {
+            var report = _pimsService.ProjectReport.Get(id);
+            return new JsonResult(_mapper.Map<ProjectReportModel>(report));
+        }
+
+        /// <summary>
+        /// Get all of the project reports.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet()]
+        [HasPermission(Permissions.AdminProjects)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ProjectReportModel[]), 200)]
+        [ProducesResponseType(typeof(ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] { "project" })]
+        public IActionResult GetProjectReports()
+        {
+            var report = _pimsService.ProjectReport.GetAll();
+            return new JsonResult(_mapper.Map<ProjectReportModel[]>(report));
+        }
+
+        /// <summary>
+        /// Get all of the project snapshots for the specified report id.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("snapshots/{reportId:int}")]
+        [HasPermission(Permissions.AdminProjects)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(ProjectSnapshotModel[]), 200)]
+        [ProducesResponseType(typeof(ErrorResponseModel), 400)]
+        [SwaggerOperation(Tags = new[] { "project" })]
+        public IActionResult GetProjectReportSnapshots(int reportId)
```

Comment:
To me it made sense to group some of this snapshot apis with the report apis since all of the snapshot data being returned is driven by the reports table.

Reply:
I agree, although you still may want a separate controller in the Reports area.